components/omega/doc/design/OmegaV1GoverningEqns.md

```diff
@@ -0,0 +1,862 @@
+
+$$
+p^{top}(x,y) =  p^{surf}(x,y) + \int_{z^{top}(x,y)}^{z^{surf}(x,y)} \rho g dz
+$$ (def-p-surf)
+
+where $z^{surf}$ is the sea surface height and $p^{surf}$ is the surface pressure at $z^{surf}$ imposed by the atmosphere or floating ice. Note that pressure increases with depth.  This means that positive $p$ points downward, so that the $top$ and $bot$ extents of the integration limits are flipped in [](#depth-pressure-integral-conversion).
+
+For any three-dimensional quantity $\varphi(x,y,z,t)$, the mass-thickness-averaged quantity in layer $k$ is defined as
+
+$$
+\varphi_k(x,y,t)
+\equiv \frac{\int_{z_k^{bot}}^{z_k^{top}} \rho \varphi dz}{\int_{z_k^{bot}}^{z_k^{top}} \rho dz}
+= \frac{\int_{z_k^{bot}}^{z_k^{top}} \rho \varphi dz}{h_k}
+$$(def-mass-thickness-average)
+
+At this point our derivation has not made any assumptions about density, and may be used for both Boussinesq and non-Boussinesq fluids. A Boussinesq derivation would now assume small variations in density and replace $\rho(x,y,z,t)$ with a constant $\rho_0$ everywhere but the pressure gradient coefficient. In that case $\rho$ divides out in [](#def-mass-thickness-average) the Boussinesq layer quantities would simply be thickness-weighted averages.
+
+We can now derive the layered equations. Integrate the continuous equations [](continuous-momentum-final), [](continuous-mass-final), [](continuous-tracer-final) in $z$ from $z_k^{bot}$ to $z_k^{top}$,
+
+$$
+\frac{\partial }{\partial t} \int_{z_k^{bot}}^{z_k^{top}} \rho {\bf u} dz
++  \int_{z_k^{bot}}^{z_k^{top}} \nabla \cdot (\rho \mathbf{u} \otimes \mathbf{u}) dz
++  \int_{z_k^{bot}}^{z_k^{top}} \frac{\partial }{\partial z} \left(  \rho w{\bf u} \right) dz
++  \int_{z_k^{bot}}^{z_k^{top}} 
+ f \rho {\bf u}^\perp dz
+  =  \int_{z_k^{bot}}^{z_k^{top}}  \left[- \nabla_z p 
+  - \rho \nabla_z \Phi 
++ \rho {\bf D}^u + \rho {\bf F}^u \right] dz
+$$ (z-integration-momentum)
+
+$$
+\frac{\partial }{\partial t}  \int_{z_k^{bot}}^{z_k^{top}} \rho dz
+ +\nabla_z \cdot \left( \int_{z_k^{bot}}^{z_k^{top}} \rho  {\bf u} dz \right)
+ + \int_{z_k^{bot}}^{z_k^{top}}\frac{\partial}{\partial z} \left(  \rho  w \right) dz
+= 0
+$$ (z-integration-mass)
+
+$$
+\frac{\partial }{\partial t} \int_{z_k^{bot}}^{z_k^{top}} \rho \varphi  dz
+ + \nabla_z \cdot \left( \int_{z_k^{bot}}^{z_k^{top}} \rho \varphi {\bf u} dz  \right)
+ + \int_{z_k^{bot}}^{z_k^{top}} \frac{\partial}{\partial z} \left(  \rho \varphi w \right) dz
+= \int_{z_k^{bot}}^{z_k^{top}} \left( D^\varphi + Q^\varphi \right) dz
+$$ (z-integration-tracers)
+ 
+This results in conservation equations that are valid over the layer. The momentum variables are simply vertically averaged. Tracer variables are vertically mass-averaged, as defined in [](def-mass-thickness-average).
+
+In order to deal with nonlinear terms where we take the integrals of products, we may assume that the variables are piecewise constant in the vertical within each layer, i.e.
```

Comment:
This approach works for some terms but not others.  For example, if you assume that velocity is constant in the vertical, you will not get the expected $w \partial \mathbf{u} / \partial z$ term in the velocity equation and you will not get agreement on what the boundary condition is between layers in the momentum equation (where all terms are still multiplied by $h_k$): one layer thinks $u_k$ should be involved, while the one above thinks $u_{k-1}$ should be there.